electrumsv/tests/test_wallet.py

```diff
@@ -550,63 +610,100 @@ def test_legacy_wallet_loading(mock_wallet_app_state, storage_info: WalletStorag
-        try:
-            wallet = Wallet(storage)
-        except FileNotFoundError as e:
-            if sys.version_info[:3] >= (3, 8, 0):
-                msg = "Could not find module 'libusb-1.0.dll' (or one of its dependencies)."
-                if msg in e.args[0]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-        except OSError as e:
-            if sys.version_info[:3] < (3, 8, 0):
-                if "The specified module could not be found" in e.args[1]:
-                    pytest.xfail("libusb DLL could not be found")
-                    return
-            raise e
-
-    # Store any pre-password update related data to compare against post-password update data.
-    prv_keypairs: dict[str, str] = {}
-    if "imported" == expected_type and "privkey" in wallet_filename:
-        assert len(wallet.get_accounts()) == 2
-        private_key_account = cast(ImportedPrivkeyAccount,
-            [ entry for entry in wallet.get_accounts() if not entry.is_petty_cash() ][0])
-        private_key_keystore = cast(Imported_KeyStore, private_key_account.get_keystore())
-        # Pre-decrypt the prv for later comparison so the initial password is not needed there.
-        for public_key, encrypted_prv in private_key_keystore._keypairs.items():
-            prv_keypairs[public_key.to_hex()] = pw_decode(encrypted_prv, initial_password)
-
-    password = "654321"
-    future, update_completion_event = wallet.update_password(initial_password, password)
+        wallet = load_wallet(storage)
+        assert wallet is not None
+
+    # Password Change should cleanly update all cached & database level encrypted data
+    new_password = "654321"
+    future, update_completion_event = wallet.update_password(initial_password, new_password)
     # Wait for the database update to finish.
     future.result(5)
     # Wait for the done callback to finish.
     update_completion_event.wait()
 
+    password_token = wallet.get_storage().get('password-token')
+    assert pw_decode(password_token, new_password) is not None
+
+    encrypted_api_keys: list[str] = []
+    server: NewServer
+    for server in wallet._servers.values():
+        server_row = server.database_rows[None]
+        if server_row is not None and server_row.encrypted_api_key is not None:
+            decoded_value = pw_decode(server.database_rows[None].encrypted_api_key,
+                new_password)
+            encrypted_api_keys.append(decoded_value)
+    if len(encrypted_api_keys):
+        test_state.count_of_server_api_key_reencryptions_cache += 1
+
+    master_key_xprvs: list[str] = []
+    master_key_seeds: list[str] = []
+    master_key_passphrases: list[str] = []
+    for keystore in wallet.get_keystores():
+        data = keystore.to_derivation_data()
+        for entry_name in ("seed", "passphrase", "xprv"):
+            entry_value = cast(str | None, data.get(entry_name, None))
```

Comment:
Different wallet files have different keystores with different derivation data. We want the counts to reflect that we have handled these different cases, so that we know that we have tested them. Your counts throw all that away and do not meet the requirements. We have no idea if we tested correct updates for a wallet with an old keystore, or a new one or a multi-signature wallet with signing data. Instead maybe from your counts we could have, but who knows. Please fix this.